### PR TITLE
[release-8.1] Upgrade jsonrpc to v1.3.0 and add epoll flags to MHD methods

### PR DESCRIPTION
**Pull request by sunny-teo9000**
## Description
Upgrade JSONRPC and add new MHD flags
Add logs to safehttpserver

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [x] small-scale cloud test

### CODE REVIEW

File: src/depends/safeserver/safehttpserver.cpp [hunk 1]
```diff
@@ -34,157 +35,197 @@ SafeHttpServer::SafeHttpServer(int port, const std::string &sslcert, const std::
...
+#if defined(MHD_USE_EPOLL_INTERNALLY)
+      mhd_flags = MHD_USE_EPOLL_INTERNALLY;
+#else
+      mhd_flags = MHD_USE_EPOLL_INTERNALLY_LINUX_ONLY;
+#endif
+    else if (has_poll)
+      mhd_flags = MHD_USE_POLL_INTERNALLY;
+
+
+    LOG_GENERAL(INFO, "Has Epoll: " << has_epoll << ", has poll: " << has_poll << ", mhd_flag: " << mhd_flags);
```
Comment (KaustubhShamshery) [inline src/depends/safeserver/safehttpserver.cpp:77]: possible debug flag, can remove?

File: src/depends/safeserver/safehttpserver.cpp [hunk 2]
```diff
@@ -34,157 +35,197 @@ SafeHttpServer::SafeHttpServer(int port, const std::string &sslcert, const std::
...
-        delete client_connection;
-    }
-    *con_cls = NULL;
-    
+int SafeHttpServer::callback(void *cls, MHD_Connection *connection, const char *url,
+                         const char *method, const char *version,
+                         const char *upload_data, size_t *upload_data_size,
+                         void **con_cls) {
+  LOG_MARKER();
+  LOG_GENERAL(INFO, "Callback started for url: " << url);
```
Comment (KaustubhShamshery) [inline src/depends/safeserver/safehttpserver.cpp:175]: possible debug flag, can remove

File: src/depends/safeserver/safehttpserver.cpp [hunk 3]
```diff
@@ -34,157 +35,197 @@ SafeHttpServer::SafeHttpServer(int port, const std::string &sslcert, const std::
...
+    struct mhd_coninfo *client_connection = new mhd_coninfo;
+    client_connection->connection = connection;
+    client_connection->server = static_cast<SafeHttpServer *>(cls);
+    *con_cls = client_connection;
     return MHD_YES;
+  }
+  struct mhd_coninfo *client_connection =
+      static_cast<struct mhd_coninfo *>(*con_cls);
+  
+  // LOG_GENERAL(INFO, "Callback for: " << client_connection->request.str());
```
Comment (KaustubhShamshery) [inline src/depends/safeserver/safehttpserver.cpp:188]: clean up comments

**Review by joel-zilliqa — APPROVED** [review]
LGTM, deployed the pod and tested, the 502 issue is no longer able to be reproduced with the changes

**Review by KaustubhShamshery — APPROVED** [review]

**Pull request merged**